index.js

```diff
@@ -1,3 +1,4 @@
+// Escaping/unescaping of "&" must happen first and last respectively in order to avoid escaping/unescaping it twice
```

Comment:
I would put a separate comment on each of these functions to make it perfectly clear.